common/flatpak-context.c

```diff
@@ -2573,7 +2593,12 @@ flatpak_context_append_bwrap_filesystem (FlatpakContext  *context,
       flatpak_bwrap_add_args_data (bwrap, "xdg-config-dirs",
                                    xdg_dirs_conf->str, xdg_dirs_conf->len, path, NULL);
     }
+}
 
-  if (exports_out)
-    *exports_out = g_steal_pointer (&exports);
+gboolean
+flatpak_context_get_shareable_ipc (FlatpakContext *context)
+{
+  /* TODO: Should this be opt-in per-app? It isn't really an
+   * --allow/--disallow feature. */
```

Comment:
I started to make this a feature flag, `features=shareable-tmp;shareable-dev-shm;` - but that didn't feel quite right, because this isn't really a *permission* in the same way as something like `--allow=bluetooth`. Giving Steam this flag gives it more control over what its subsandboxes see, but doesn't give it any more access to the host system.

At the same time, making it unconditional doesn't feel right, because ideally we'd only be creating and destroying host-visible temp directories for apps that actually need them. WDYT?

Reply:
I'd say it better be unconditional. Probably there are no drawbacks except for some users may not like seeing the directories we create?

Reply:
Making these shareable between *any instance* feels a bit more like a feature flag to me.

You could argue that what we're `--allow`'ing is "can create a directory on the host that persists for a time after the app has gone away".

Reply:
> except for some users may not like seeing the directories we create

@alexlarsson seemed to think that this is potentially a big deal: "most" Flatpak apps won't need this, so "most" Flatpak apps will just be creating and destroying an empty directory, which users will whine about.

Reply:
> @alexlarsson seemed to think that this is potentially a big deal:

Then why creating `$XDG_RUNTIME_DIR/app/$FLATPAK_ID` wasn't big deal? If you bind `$XDG_RUNTIME_DIR/app/$FLATPAK_ID` to `/tmp` you wouldn't even need to create anything new and nobody will have a chance to complain if didn't already.

> Making these shareable between any instance feels a bit more like a feature flag to me.

Making many apps work without hacks shouldn't be a feature but necessity. I don't see what not sharing /tmp or /dev/shm gives except breaking things in default case. I agree that for `--sandbox` case they shouldn't be shared.

Reply:
>Flatpak apps will just be creating and destroying an empty directory, which users will whine about

Honestly, I fail to see how this is a big deal.

Reply:
> Then why creating `$XDG_RUNTIME_DIR/app/$FLATPAK_ID` wasn't big deal?

Novice users with just enough knowledge to be dangerous don't look there, so they don't notice; and users with advanced enough knowledge to look there are aware of just how "cheap" it is to create and destroy directories on a tmpfs, so they don't care.

Reply:
I'm not sure we're on the same page but just few comments above you said users will whine about those created dirs and now you say creating them is so cheap that users don't care.

If we both agree that `$XDG_RUNTIME_DIR/app/$FLATPAK_ID` isn't big deal then I don't see the problem here.

Reply:
I'm only relaying concerns that Alex raised on IRC. If maintainers can agree that I should do or not do a particular thing, I'll do that.

Until then, I'm going to go with implementing and testing the option that involves a bit more code, so that if maintainers want it, they can keep it, and if maintainers tell me to drop it, I can respond to that feedback quickly.

Creating empty directories on a tmpfs is so cheap that it won't *actually* cause a significant performance impact for anyone, but I think @alexlarsson was concerned that if we create and destroy lots of empty directories in `/tmp`, which is a user-visible location that semi-experienced users will look at, then they will *think* we are causing a performance problem - this is more about perception than reality.

Reply:
I've split the conditionality into a separate commit, so that it's trivial to either keep or drop.